### DISCUSSION TITLE
Add Specific Advancement as requirement in Rank Up System

### DISCUSSION THREAD

**Discussion by limimc** [Zrips/CMI#6597] [Feature Requests]
Hi,

We are currently configuring our rankup requirements and one of our rank is focused on exploration. It would be nice if we have an option to add specific advancement as requirement to rankup. For example; Player must have **Adventuring Time** (Discover every biome ) advancement to rank up.

https://minecraft.fandom.com/wiki/Advancement

Example format:

```
AdvancementRequirement:
- "advancement:adventuring_time"
- "advancement:spyglass_at_dragon"
```

**Comment by mrfloris** [comment]
I would personally already be happy if we can see if they've completed an amount of advancements, but having certain completed in order to rankup would be a cool addition. +1

**Comment by STSNOC** [comment]
I would love this feature as well so I could ditch another plugin that does advancements, but won't support all vanilla stats. So right now I've had to lash CMI and Autorank together to achieve a stat/advancement based rank.
